lib/jsdom/browser/Window.js

```diff
@@ -412,3 +411,9 @@ function stopAllTimers(window) {
   });
   window.__timers = [];
 }
+
+function fixUrlForBuggyXhr(url) {
+  // node-XMLHttpRequest doesn't properly handle file URLs. It only accepts file://C:/..., not file:///C:/...
+  // See https://github.com/tmpvar/jsdom/pull/1180
+  return url.replace(/^file:\/\/\//, "file://");
```

Comment:
Are you testing on darwin/linux? I'm thinking that actually 3 slashes would still work on darwin/linux and you need to specifically look for `/^file:\/\/\/([a-zA-Z]:)/`, since the url `file:///foo/bar.txt` would still be valid on darwin and linux as it would yield the path `/foo/bar.txt`. 

The problem case is when sending the url `/c:/bar.txt` on windows vs. `c:/bar.txt`, which the former node-xhr project fails on.